advanced_functionality/pytorch_extending_our_containers/container/cifar10/cifar10.py

```diff
@@ -0,0 +1,171 @@
+                                              shuffle=False, num_workers=args.workers)
+
+    logger.info("Model loaded")
+    model = Net()
+
+    if torch.cuda.device_count() > 1:
+        logger.info("Gpu count: {}".format(torch.cuda.device_count()))
+        model = nn.DataParallel(model)
+
+    model = model.to(device)
+
+    criterion = nn.CrossEntropyLoss().to(device)
+    optimizer = torch.optim.SGD(model.parameters(), lr=args.lr, momentum=args.momentum)
+
+    for epoch in range(0, args.epochs):
+        running_loss = 0.0
+        for i, data in enumerate(train_loader):
+            # get the inputs
+            inputs, labels = data
+            inputs, labels = inputs.to(device), labels.to(device)
+
+            # zero the parameter gradients
+            optimizer.zero_grad()
+
+            # forward + backward + optimize
+            outputs = model(inputs)
+            loss = criterion(outputs, labels)
+            loss.backward()
+            optimizer.step()
+
+            # print statistics
+            running_loss += loss.item()
+            if i % 2000 == 1999:  # print every 2000 mini-batches
+                print('[%d, %5d] loss: %.3f' %
+                      (epoch + 1, i + 1, running_loss / 2000))
+                running_loss = 0.0
+    print('Finished Training')
+    return _save_model(model, args.model_dir)
+
+
+def _save_model(model, model_dir):
+    logger.info("Saving the model.")
+    path = os.path.join(model_dir, 'model.pth')
+    # recommended way from http://pytorch.org/docs/master/notes/serialization.html
+    torch.save(model.cpu().state_dict(), path)
+
+
+def model_fn(model_dir):
+    logger.info('model_fn')
+    device = "cuda" if torch.cuda.is_available() else "cpu"
+    model = Net()
+    if torch.cuda.device_count() > 1:
+        logger.info("Gpu count: {}".format(torch.cuda.device_count()))
+        model = nn.DataParallel(model)
+
+    with open(os.path.join(model_dir, 'model.pth'), 'rb') as f:
+        model.load_state_dict(torch.load(f))
+    return model.to(device)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument('--workers', type=int, default=2, metavar='W',
+                        help='number of data loading workers (default: 2)')
+    parser.add_argument('--epochs', type=int, default=2, metavar='E',
+                        help='number of total epochs to run (default: 2)')
+    parser.add_argument('--batch-size', type=int, default=4, metavar='BS',
+                        help='batch size (default: 4)')
+    parser.add_argument('--lr', type=float, default=0.001, metavar='LR',
+                        help='initial learning rate (default: 0.001)')
+    parser.add_argument('--momentum', type=float, default=0.9, metavar='M', help='momentum (default: 0.9)')
+    parser.add_argument('--dist-backend', type=str, default='gloo', help='distributed backend (default: gloo)')
+
+    env = sagemaker_containers.training_env()
```

Comment:
You are using the training env here but you are not explaining how it works anywhere. Consider add a little burb about this method in the notebook and link to this https://github.com/aws/sagemaker-containers/blob/master/src/sagemaker_containers/_env.py#L385


Another option, is to use env vars and instead and link to this docs here https://github.com/aws/sagemaker-containers#how-a-script-is-executed-inside-the-container

I believe that Leo likes more the second option.

Reply:
Swapped to env vars.